### PR TITLE
feat: use dotnet 8 SDK

### PR DESCRIPTION
**Pull request by vaind**
Updates to .NET 8 SDK which is currently on RC.2 (needed to release sentry-dotnet v4.0.0.alpha)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -27,19 +27,19 @@ ENV DEBIAN_FRONTEND=noninteractive \
...
+  git \
+  python3-packaging \
+  ruby-full \
+  twine \
+  jq \
+  unzip \
+  openjdk-11-jdk \
+  maven \
   && apt-get clean \
   && rm -rf /var/lib/apt/lists/*
```
Comment (asottile-sentry) [inline Dockerfile:44]: please don't make unrelated changes in your PR

**Review by asottile-sentry — APPROVED** [review]

**Comment by vaind** [comment]
sorry about the broken build, should be fixed now

**Comment by vaind** [comment]
Build works fine. I can't merge in this repo so if someone else wants to do that...
